File: ConsumerFlink/src/main/java/Dto/CrimeRecord.java
package Dto;

import lombok.Data;
import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonFormat;
import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.annotation.JsonProperty;


//import java.time.LocalDateTime;
//import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule.LocalDateTime


@Data
public class CrimeRecord {

    @JsonProperty("ID")
    private int id;

    @JsonProperty("Case_Number")
    private String caseNumber;

    @JsonProperty("Date")
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private String date;

    @JsonProperty("Block")
    private String block;

    @JsonProperty("IUCR")  // Add @JsonProperty to map the field properly
    private String iucr;

    @JsonProperty("Primary_Type")
    private String primaryType;

    @JsonProperty("Description")
    private String description;

    @JsonProperty("Location_Description")
    private String locationDescription;

    @JsonProperty("Arrest")
    private boolean arrest;

    @JsonProperty("Domestic")
    private boolean domestic;

    @JsonProperty("Beat")
    private int beat;

    @JsonProperty("District")
    private Integer district;

    @JsonProperty("Ward")
    private Integer ward;

    @JsonProperty("Community_Area")
    private Integer communityArea;

    @JsonProperty("FBI_Code")
    private String fbicode;

    @JsonProperty("X_Coordinate")
    private Integer xCoordinate;

    @JsonProperty("Y_Coordinate")
    private Integer yCoordinate;

    @JsonProperty("Year")
    private int year;

    @JsonProperty("Updated_On")
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private String updatedOn;

    @JsonProperty("Latitude")
    private Double latitude;

    @JsonProperty("Longitude")
    private Double longitude;

    @JsonProperty("Location")
    private String location;
}

File: app/CrimeRecord.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional



class CrimeRecord(BaseModel):
    ID: int
    Case_Number: str
    Date: datetime
    Block: str
    IUCR: str
    Primary_Type: str
    Description: str
    Location_Description: Optional[str]
    Arrest: bool
    Domestic: bool
    Beat: int
    District: Optional[int]
    Ward: Optional[int]
    Community_Area: Optional[int]
    FBI_Code: str
    X_Coordinate: Optional[int]
    Y_Coordinate: Optional[int]
    Year: int
    Updated_On: datetime
    Latitude: Optional[float]
    Longitude: Optional[float]
    Location: Optional[str]

    class Config:
        arbitrary_types_allowed = True
